runtime/opt/taupage/runtime/Docker.py

```diff
@@ -186,29 +184,14 @@ def registry_login(config: dict, registry: str):
         logging.warning('Docker registry seems not to be Pier One, skipping OAuth login')
         return
     pierone_url = 'https://{}'.format(registry)
-    token_url = config.get('token_service_url')
 
-    if not token_url:
-        logging.warning('No token service URL configured in Taupage YAML ("token_service_url" property)')
-        return
-
-    path = os.path.join(CREDENTIALS_DIR, 'user.json')
-
-    while not os.path.exists(path):
-        logging.info('Waiting for berry to download OAuth credentials to {}..'.format(path))
-        time.sleep(5)
-
-    with open(path) as fd:
-        credentials = json.load(fd)
-
-    user = credentials.get('application_username')
-    passwd = credentials.get('application_password')
+    token = get_token(config, 'pierone', ['uid'])
 
-    if not user or not passwd:
-        logging.warning('Invalid OAuth credentials: application user and/or password missing in %s', path)
+    if not token or 'access_token' not in token:
+        logging.warning('Missing OAuth token for Pier One login')
```

Comment:
user's (in a different organization) might still use a Pier One configured without OAuth, so better continue...